independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/PublisherResponseHandler.java

```diff
@@ -312,38 +317,39 @@ private boolean requiresChunkedStream(MediaType mediaType) {
                 || mediaType.isCompatible(RestMediaType.APPLICATION_STREAM_JSON_TYPE);
     }
 
+    @SuppressWarnings("rawtypes")
     private void handleChunkedStreaming(ResteasyReactiveRequestContext requestContext, Publisher<?> result, boolean json) {
         long demand = 1L;
-        if (result instanceof RestMulti.SyncRestMulti) {
-            RestMulti.SyncRestMulti rest = (RestMulti.SyncRestMulti) result;
+        if (result instanceof RestMulti.SyncRestMulti rest) {
             demand = rest.getDemand();
         }
         result.subscribe(
                 new StreamingMultiSubscriber(requestContext, streamingResponseCustomizers, result, json, demand, false));
     }
 
+    @SuppressWarnings("rawtypes")
     private void handleStreaming(ResteasyReactiveRequestContext requestContext, Publisher<?> result, boolean json) {
         long demand = 1L;
         boolean encodeAsJsonArray = true;
-        if (result instanceof RestMulti.SyncRestMulti) {
-            RestMulti.SyncRestMulti rest = (RestMulti.SyncRestMulti) result;
+        if (result instanceof RestMulti.SyncRestMulti rest) {
             demand = rest.getDemand();
             encodeAsJsonArray = rest.encodeAsJsonArray();
         }
         result.subscribe(new StreamingMultiSubscriber(requestContext, streamingResponseCustomizers, result, json, demand,
                 encodeAsJsonArray));
     }
 
+    @SuppressWarnings("rawtypes")
     private void handleSse(ResteasyReactiveRequestContext requestContext, Publisher<?> result) {
         long demand;
-        if (result instanceof RestMulti.SyncRestMulti) {
-            RestMulti.SyncRestMulti rest = (RestMulti.SyncRestMulti) result;
+        if (result instanceof RestMulti.SyncRestMulti rest) {
             demand = rest.getDemand();
         } else {
             demand = 1L;
         }
 
-        SseUtil.setHeaders(requestContext, requestContext.serverResponse(), streamingResponseCustomizers);
+        List<StreamingResponseCustomizer> customizers = determineCustomizers(result, true, streamingResponseCustomizers);
+        SseUtil.setHeaders(requestContext, requestContext.serverResponse(), customizers);
```

Comment:
I think we need to subscribe first and write headers on the first item, as done in `StreamingMultiSubscriber`. Otherwise customizers from `upstream` in `AsyncRestMulti` won't pick up. See test case `testReturnRestSse3`.